### PR TITLE
Improve documentation for verify_header plug module

### PR DESCRIPTION
**Pull request by edouardmenayde**
I'm new to elixir and its ecosystem thus i'm starting to learn its ins and outs including guardian. I love how easy testing is with elixir but I couldn't understand why my test couldn't pass thus I had to dig through the code to understand how the header is parsed by default. I think adding a few line of documentation could help the next guy experience.

### CODE REVIEW

**Comment by oskar1233** [comment]
IMHO the stuff you covered is more about understanding  building HTTP request with authorization header than about `verify_header` module itself. Maybe readme or even some separate  guide would be better place to write about it?

**Comment by edouardmenayde** [comment]
@oskar1233 I agree it might be best in the readme. Do I create a new PR or do I amend the existing one ?

**Comment by doomspork** [comment]
@edouardmenayde thanks for getting involved!  I agree with @edouardmenayde that this doesn't really belong in the code documentation, especially not without more context.

Perhaps we should look into adding a section to the README for now on testing best practices?

Thoughts @ueberauth/developers?

**Comment by oskar1233** [comment]
Maybe some subheading in the basics section? Just because that's what you usually do in the beginning and boosts your API understanding :)

**Comment by edouardmenayde** [comment]
Hi!
It's been some days since the last comment thus I think if there's any strong insight against adding that to the readme It would have been raised by now. I took the initiative to amend my last commit and move the documentation to the readme.

File: README.md [hunk 1]
```diff
@@ -387,6 +387,16 @@ This will record each token issued in your database, confirm it is still valid o
 
 For more in-depth documentation please see the [GuardianDb README](https://github.com/ueberauth/guardian_db/blob/master/README.md)
 
+## Best testing practises
```
Comment (yordis) [inline README.md]: Misspelling, change `practises` to `practices`

File: README.md [hunk 2]
```diff
@@ -387,6 +387,16 @@ This will record each token issued in your database, confirm it is still valid o
...
+
+### How to add the token to a request (the Phoenix way)
+```elixir
+{:ok, token, _} = encode_and_sign(resource, %{}, token_type: :access)
```
Comment (yordis) [inline README.md]: I would recommend to add a full example of the Controller so newbies do not get lost in the context.

For example, where `encode_and_sign/2`  come from?
Where should I add this code?

**Comment by edouardmenayde** [comment]
@yordis I fixed the typo and gave some more context to the test. I tried giving much more context with the controller etc. but It become quickly more of a blog post than a little hint to test guardian.

**Comment by Moggers** [comment]
This just saved me from hours of headache. I hadn't used the `Authorization` header before and didn't realize that it had to look like `authorization: <type>: <token>`.

File: README.md [hunk 3]
```diff
@@ -387,6 +392,30 @@ This will record each token issued in your database, confirm it is still valid o
...
+
+### How to add the token to a request (the Phoenix way)
+
+```elixir
+defmodule HelloWeb.AuthControllerTest do
+  use HelloWeb.ConnCase
+  import HelloWeb.Guardian
+
+ test "GET /auth/me", %{conn: conn} do
+    user = insert(:user) // See https://github.com/thoughtbot/ex_machina
```
Comment (doomspork) [inline README.md]: This needs to be `# See https://github.com/thoughtbot/ex_machina`

Comment (edouardmenayde) [reply README.md]: Oh I will correct that :)

**Comment by edouardmenayde** [comment]
@Moggers really happy my headache saved you from having one :)

**Comment by edouardmenayde** [comment]
@doomspork I fixed my mistake, this should be ok to merge :-)

**Comment by doomspork** [comment]
Thank you @edouardmenayde! 👍

**Pull request merged**